bundle/Command/TagContentByTypesCommand.php

```diff
@@ -0,0 +1,198 @@
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Exception\InvalidOptionException;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+
+use function array_filter;
+use function array_map;
+use function array_unique;
+use function array_values;
+use function explode;
+use function sprintf;
+use function trim;
+
+final class TagContentByTypesCommand extends Command
+{
+    private SymfonyStyle $style;
+    private InputInterface $input;
+
+    public function __construct(
+        private Repository              $repository,
+        private TagsService             $tagsService,
+        private ConfigResolverInterface $configResolver,
+    ) {
+        // Parent constructor call is mandatory for commands registered as services
+        parent::__construct();
+    }
+
+    protected function configure(): void
+    {
+        $this->setDescription('Add tags to content')
+            ->addOption('parent-location', null, InputOption::VALUE_REQUIRED)
+            ->addOption('content-types', null, InputOption::VALUE_REQUIRED)
+            ->addOption('tag-id', null, InputOption::VALUE_REQUIRED)
+            ->addOption('field-identifiers', null, InputOption::VALUE_OPTIONAL);
+    }
+
+    protected function initialize(InputInterface $input, OutputInterface $output): void
+    {
+        $this->input = $input;
+        $this->style = new SymfonyStyle($input, $output);
+    }
+
+    protected function execute(InputInterface $input, OutputInterface $output): int
+    {
+        $query = new Query();
+        $query->filter = new Criterion\LogicalAnd([
+            new Criterion\ParentLocationId($this->getParentLocation()),
+            new Criterion\ContentTypeIdentifier($this->getContentTypes()),
+        ]);
+
+        $batchSize = 50;
+
+        $searchResults = $this->repository->getSearchService()->findContent($query);
+        $totalResults = $searchResults->totalCount;
+
+        $this->style->newLine();
+        $this->style->progressStart($totalResults);
+
+        $this->repository->beginTransaction();
+
+        for ($offset = 0; $offset < $totalResults; $offset += $batchSize) {
+            $query->offset = $offset;
+            $query->limit = $batchSize;
+            $searchResults = $this->repository->getSearchService()->findContent($query);
+
+            if ($this->input->getOption('field-identifiers') === null) {
+                $fieldIdentifiers = $this->configResolver->getParameter('tag_command_default_field_identifiers', 'ngsite');
+            } else {
+                $fieldIdentifiers = $this->getFieldIdentifiers();
+            }
+
+            $this->repository->beginTransaction();
+
+            foreach ($searchResults->searchHits as $searchHit) {
+                $result = $this->repository->sudo(
```

Comment:
sudo should be used only for parts of the code that require it, not for everything.

Reply:
fixed in 14aef55e